src/components/profile/EditAddress.jsx

```diff
@@ -0,0 +1,188 @@
+import React, { useEffect, useState } from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { fetchUserProfile, updateUserProfileData } from "../../redux/reducers/profileSlice";
+import ProfileMenu from "./ProfileMenu";
+import { showErrorMessage, showSuccessMessage } from "../../utils/toast";
+
+export default function UpdateAddress () {
+  const dispatch = useDispatch();
+  const { user, isLoading, isError } = useSelector((state) => state.profile);
+
+  useEffect(() => {
+    dispatch(fetchUserProfile());
+  }, [dispatch]);
+
+  const [email, setEmail] = useState("");
+  const [phoneNo, setPhoneNo] = useState("");
+  const [province, setProvince] = useState("");
+  const [district, setDistrict] = useState("");
+  const [street, setStreet] = useState("");
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const updatedBillingAddress = {
+      email,
+      phoneNo,
+      province,
+      district,
+      street,
+    };
+    dispatch(updateUserProfileData(updatedBillingAddress))
+      .then(() => {
+        showSuccessMessage("User profile updated successfully");
+        console.log("User profile updated successfully");
+      })
+      .catch((error) => {
+        showErrorMessage("Error updating user profile");
+        console.error("Error updating user profile:", error);
+      });
+  };
+
+  useEffect(() => {
+    const address = user?.payload?.user_details.billingAddress || {};
+    setEmail(address.email);
+    setPhoneNo(address.phoneNo);
+    setProvince(address.province);
+    setDistrict(address.district);
+    setStreet(address.street);
+  }, [user]);
+
```

Comment:
This is too much , you need to use react-hook-form for handling the you form functionalities instead of creating state for every task